apiclient/rancher_api/managers.py

```diff
@@ -0,0 +1,190 @@
+    # server-version
+    PATH_fmt = "apis/management.cattle.io/v3/settings/{name}"
+    # "v1/harvesterhci.io.settings/{name}"
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.PATH_fmt.format(name=name))
+
+
+class MgmtClusterManager(BaseManager):
+    CREATE_fmt = "v1/provisioning.cattle.io.clusters"
+    GET_fmt = "apis/provisioning.cattle.io/v1/namespaces/{ns}/clusters/{uid}"
+    DELETE_fmt = "v1/provisioning.cattle.io.clusters/{ns}/{uid}"
+
+    def create_data(self, cluster_name, spec={}, labels={}):
+        return {
+            "type": "provisioning.cattle.io.cluster",
+            "metadata": {
+                "namespace": FLEET_DEFAULT_NAMESPACE,
+                "labels": labels,
+                "name": cluster_name
+            },
+            "spec": spec
+        }
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.GET_fmt.format(uid=name, ns=FLEET_DEFAULT_NAMESPACE), raw=raw)
+
+    def create(self, name, spec={}, labels={}, raw=False):
+        data = self.create_data(name, spec=spec, labels=labels)
+        return self._create(self.CREATE_fmt, json=data, raw=raw)
+
+    def delete(self, name, *, raw=False):
+        return self._delete(self.DELETE_fmt.format(uid=name, ns=FLEET_DEFAULT_NAMESPACE))
+
+
+class ClusterRegistrationTokenManager(BaseManager):
+    PATH_fmt = "v3/clusterRegistrationTokens/{uid}:default-token"
+
+    def get(self, name="", *, raw=False):
+        return self._get(self.PATH_fmt.format(uid=name), raw=raw)
+
+
+class CloudCredentialManager(BaseManager):
+    PATH_fmt = "v3/cloudcredentials"
+
+    def create_data(self, name, cluster_id, kubeconfig):
+        return {
+            "type": "provisioning.cattle.io/cloud-credential",
+            "metadata": {
+                "generateName": "cc-",
+                "namespace": FLEET_DEFAULT_NAMESPACE
+            },
+            "_name": name,
+            "annotations": {
+                "provisioning.cattle.io/driver": "harvester"
+            },
+            "harvestercredentialConfig": {
+                "clusterType": "imported",
+                "clusterId": cluster_id,
+                "kubeconfigContent": kubeconfig
+            },
+            "_type": "provisioning.cattle.io/cloud-credential",
+            "name": name
+        }
+
+    def create(self, name, cluster_id, kubeconfig, *, raw=False):
+        data = self.create_data(name, cluster_id, kubeconfig)
+        return self._create(self.PATH_fmt, json=data, raw=raw)
+
+    def list(self, *, raw=False):
+        return self._get(self.PATH_fmt, raw=raw)
+
+
+class KubeConfigManager(BaseManager):
+    PATH_fmt = "k8s/clusters/{cluster_id}/v1/harvester/kubeconfig"
+
+    def create_data(self, name):
+        return {
+            "clusterRoleName": "harvesterhci.io:cloudprovider",
+            "namespace": DEFAULT_NAMESPACE,
+            "serviceAccountName": name
+        }
+
+    def create(self, name, cluster_id, *, raw=True):
```

Comment:
update the BaseManager, then the option `raw` will align other methods.

https://github.com/harvester/tests/blob/5082bc97a20b7d40446266330dc980fffeb3af1a/apiclient/harvester_api/managers.py#L32-L45